### ISSUE TITLE
Logout with SAML throws Whoops! on logout

### ISSUE THREAD

**Issue by niobis** [grokability/snipe-it#10706]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

I enabled SAML-Login with our onPrem installation and AzureAD as IDP. The login works perfectly, but on logout snipe-it throws the error:

```
Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException thrown with message "The GET method is not supported for this route. Supported methods: POST."
```

### Reproduction steps

1. Login to snipe-it with SAML-login
2. Click logout on user
3. Above error message occurs.

### Expected behavior

A clean logout/redirect to the login page

### Snipe-IT Version

v5.3.10

### Operating System

Ubuntu 20.04.4 LTS

### Web Server

Apache

### PHP Version

7.4.3

### Operating System

MacOS

### Browser

Google Chrome

### Version

98

### Error messages

```shell
Stacktrace:
#35 Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/RouteCollection.php:256
#34 Illuminate\Routing\RouteCollection:methodNotAllowed in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/RouteCollection.php:242
#33 Illuminate\Routing\RouteCollection:getRouteForMethods in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/RouteCollection.php:176
#32 Illuminate\Routing\RouteCollection:match in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php:635
#31 Illuminate\Routing\Router:findRoute in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php:624
#30 Illuminate\Routing\Router:dispatchToRoute in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Routing/Router.php:613
#29 Illuminate\Routing\Router:dispatch in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:170
#28 Illuminate\Foundation\Http\Kernel:Illuminate\Foundation\Http\{closure} in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:130
#27 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php:67
#26 Barryvdh\Debugbar\Middleware\InjectDebugbar:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#25 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/app/Http/Middleware/PreventBackHistory.php:23
#24 App\Http\Middleware\PreventBackHistory:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#23 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/app/Http/Middleware/SecurityHeaders.php:26
#22 App\Http\Middleware\SecurityHeaders:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#21 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21
#20 Illuminate\Foundation\Http\Middleware\TransformsRequest:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#19 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/app/Http/Middleware/CheckForDebug.php:25
#18 App\Http\Middleware\CheckForDebug:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#17 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/app/Http/Middleware/CheckForSetup.php:26
#16 App\Http\Middleware\CheckForSetup:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#15 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/fideloper/proxy/src/TrustProxies.php:57
#14 Fideloper\Proxy\TrustProxies:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#13 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php:49
#12 Illuminate\View\Middleware\ShareErrorsFromSession:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#11 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:56
#10 Illuminate\Session\Middleware\StartSession:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#9 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php:63
#8 Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#7 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/app/Http/Middleware/NoSessionStore.php:27
#6 App\Http\Middleware\NoSessionStore:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#5 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/barryvdh/laravel-cors/src/HandlePreflight.php:29
#4 Barryvdh\Cors\HandlePreflight:handle in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:171
#3 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:105
#2 Illuminate\Pipeline\Pipeline:then in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:145
#1 Illuminate\Foundation\Http\Kernel:sendRequestThroughRouter in /var/www/html/snipe-it/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:110
#0 Illuminate\Foundation\Http\Kernel:handle in /var/www/html/snipe-it/public/index.php:58
```

### Additional context

git installation, updated via `php upgrade.php`
Error did occur with previous snipe-it version as well

```
[PHP Modules]
bcmath
calendar
Core
ctype
curl
date
dom
exif
FFI
fileinfo
filter
ftp
gd
gettext
hash
iconv
igbinary
imagick
json
ldap
libxml
mbstring
memcached
msgpack
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_pgsql
pgsql
Phar
posix
readline
Reflection
session
shmop
SimpleXML
soap
sockets
sodium
SPL
standard
sysvmsg
sysvsem
sysvshm
tokenizer
xml
xmlreader
xmlwriter
xsl
Zend OPcache
zip
zlib

[Zend Modules]
Zend OPcache
```

**Comment by snipe** [comment]
@uberbrady I believe we were discussing  better error handling here last week. Can you take a look? This can be for develop if necessary.

**Comment by chauyuen** [comment]
Hi,in fact i am facing the same issue when i using keycloak with Snipe-it.

**Comment by dritter9** [comment]
@niobis @snipe @uberbrady the issue is caused in SamlController@sls, when you use redirect, you can't redirect from get to post (route logout), even if you use route() method on redirect helper, so you get the MethodNotAllowedHttpException.

When you change the return of the sls function in App\Http\Controllers\Auth\SamlController.php to the follows, everything works fine
`return redirect()->to($sloUrl)->with(auth()->logout());`

Azure IdP Configuration (as described in https://snipe-it.readme.io/docs/saml)
<img width="784" alt="Bildschirmfoto 2022-05-01 um 02 02 43" src="https://user-images.githubusercontent.com/60943273/166126654-059a7404-70d1-4059-8cda-513433fca40a.png">

Snipe-IT SAML Configuration
<img width="756" alt="Bildschirmfoto 2022-05-01 um 02 22 27" src="https://user-images.githubusercontent.com/60943273/166127014-49498f74-504d-4423-81c6-c310fac4e737.png">


Could any body merge this, if thats a good solution?

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/saml">SAML</a></strong></div><div>Configuration guidelines for SAML Single-Sign On (SSO) support</div></blockquote>

**Comment by johnson-yi** [comment]
From investigating I think this is happening due to this change https://github.com/snipe/snipe-it/pull/10283 for removing the GET logout route.

Can you try to add the route back and test from your end?

master - https://github.com/snipe/snipe-it/blob/master/routes/web.php
develop - https://github.com/snipe/snipe-it/blob/develop/routes/web.php

```
    Route::get(
        'logout',
        [
            'as' => 'logout',
            'uses' => 'Auth\LoginController@logout' ]
    );
```


<blockquote><img src="https://repository-images.githubusercontent.com/14346979/164a6580-7d79-11e9-865f-eae87593e11a" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/snipe/snipe-it">snipe-it/web.php at master · snipe/snipe-it</a></strong></div><div>A free open source IT asset/license management system - snipe-it/web.php at master · snipe/snipe-it</div></blockquote>
<blockquote><img src="https://repository-images.githubusercontent.com/14346979/164a6580-7d79-11e9-865f-eae87593e11a" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/snipe/snipe-it">snipe-it/web.php at develop · snipe/snipe-it</a></strong></div><div>A free open source IT asset/license management system - snipe-it/web.php at develop · snipe/snipe-it</div></blockquote>

**Comment by dritter9** [comment]
@johnson-yi adding the logout route manually as GET back again, solved the problem in docker images v5.4.3 and latest. I think that would be the better way, so custom logout url if set in config will proceed too.

I got an LogicException when use `php artisan route:cache` in both versions, but thats not caused from adding the route back again, the application seems to fully work for me.
<img width="853" alt="Bildschirmfoto 2022-05-01 um 17 46 10" src="https://user-images.githubusercontent.com/60943273/166153619-45e42b14-b292-4937-90ce-e3f937e2e1cf.png">
.
